src/lib.rs

```diff
@@ -160,67 +184,135 @@ pub trait Get<T>: Collection {
 /// Mutably queryable collection.
 pub trait GetMut<T>: Get<T> {
 	/// Returns a mutable reference to the item stored behind the given key (if any).
-	fn get_mut(&mut self, key: T) -> Option<&mut Self::Item>;
+	fn get_mut<'a>(&'a mut self, key: T) -> Option<Self::ItemMut<'a>>;
+}
+
+/// Guarded queryable collection.
+pub trait GetLock<T>: Get<T> {
```

Comment:
So you want to define interfaces for concurrent data structures. I'm not a fan of using the term `Lock` in the name of the trait since such data structure may be implemented without any lock. Maybe `ConcurrentGet` (a bit long) or `ConstGet` (maybe ambiguous) would be more appropriate? Anyway I think introducing concurrent traits is well beyond just adding GATs and should be part of another pull request so we can discuss these details.

Reply:
> So you want to define interfaces for concurrent data structures. I'm not a fan of using the term `Lock` in the name of the trait since such data structure may be implemented without any lock. Maybe `ConcurrentGet` (a bit long) or `ConstGet` (maybe ambiguous) would be more appropriate? Anyway I think introducing concurrent traits is well beyond just adding GATs and should be part of another pull request so we can discuss these details.

Okay, I'll split this into two PRs, I'll keep this PR on GATs.